src/AzureAD.tsx

```diff
@@ -58,61 +54,53 @@ class AzureAD extends React.Component<IProps, IState> {
   constructor(props: IProps) {
     super(props);
 
-    const authenticationState = AuthenticationState.Unauthenticated;
-
     this.authProvider = this.props.provider.getAuthProvider();
-    this.authProvider.userInfoChangedCallback = this.updateState;
+    this.authProvider.onAuthenticationStateChanged = this.updateAuthenticationState;
 
-    this.state = { authenticationState };
+    this.state = { authenticationState: this.authProvider.authenticationState }
   }
 
   public render() {
     switch (this.state.authenticationState) {
       case AuthenticationState.Authenticated:
-        return this.props.authenticatedFunction(this.logout);
+        if (this.props.authenticatedFunction) {
```

Comment:
I added null checks to all of the callbacks. Sometimes you might want to use the component to only display content when the user is logged in, or out. You should be able to exclude a callback when useful.